homeassistant/components/google/manifest.json

```diff
@@ -5,6 +5,7 @@
   "dependencies": ["auth"],
   "documentation": "https://www.home-assistant.io/integrations/calendar.google/",
   "requirements": [
+    "gcal-sync==0.3.1",
     "google-api-python-client==2.38.0",
     "httplib2==0.20.4",
     "oauth2client==4.1.3"
```

Comment:
I want us to move away from oauth2client, httplib2 and google-api-python-client.

- oauth2client is deprecated.
- httplib2 is only needed as a transport requirement of google-api-python-client. We shouldn't need to pin transport requirements of the primary library we use for an api.
- google-api-python-client doesn't support asyncio natively.

Reply:
I am on the same page as you, and have been thinking about how we drop all of these. Here is what I have for you so far:

- oauth2client's device flow has no replacement as far as I know, so it would look like a native re-implementation. I am currently in talks with google dev rel about the long term of what we want oauth to look like for nest given the OOB deprecation which may include using device flow or it may include something else, so I don't have an answer for you yet on this one.

- http2lib and google-api-python-client I agree should go.  httplib2 has a direct dependency for an exception, as the moment. I would like to replace with asyncio and use those exceptions instead. my thought was that I would rewrite all of this in asyncio, fix the error handling, and get this to platinum. I'd bias towards making this work like google-nest-sdm avoiding all google libraries since I don't see them adding much.  (I'm not sold aiogoogle is better)

If you'd rather have this as a massive PR, happy to do it. However my #1 priority is usually to make reviewer life easier, so my suggestion is we  agree on a long term vision/direction then getting there incrementally.